pkg/operator/management/management_state.go

```diff
@@ -0,0 +1,69 @@
+package management
+
+import (
+	"github.com/openshift/api/operator/v1"
+)
+
+var (
+	allowOperatorUnmanagedState = true
+	allowOperatorRemovedState   = true
+)
+
+// These are for unit testing
+var (
+	getAllowedOperatorUnmanaged = func() bool {
+		return allowOperatorUnmanagedState
+	}
+	getAllowedOperatorRemovedState = func() bool {
+		return allowOperatorRemovedState
+	}
+)
+
+// SetOperatorAlwaysManaged is one time choice when an operator want to opt-out from supporting the "unmanaged" state.
+// This is a case of control plane operators or operators that are required to always run otherwise the cluster will
+// get into unstable state or critical components will stop working.
+func SetOperatorAlwaysManaged() {
+	allowOperatorUnmanagedState = false
+}
+
+// SetOperatorNotRemovable is one time choice the operator author can make to indicate the operator does not support
+// removing of his operand. This makes sense for operators like kube-apiserver where removing operand will lead to a
+// bricked, non-automatically recoverable state.
+func SetOperatorNotRemovable() {
+	allowOperatorRemovedState = false
+}
+
+// IsOperatorAlwaysManaged means the operator can't be set to unmanaged state.
+func IsOperatorAlwaysManaged() bool {
+	return !getAllowedOperatorUnmanaged()
+}
+
+// IsOperatorNotRemovable means the operator can't bet set to removed state.
+func IsOperatorNotRemovable() bool {
+	return !getAllowedOperatorRemovedState()
+}
+
+func IsOperatorUnknownState(state v1.ManagementState) bool {
+	switch state {
+	case v1.Managed, v1.Removed, v1.Unmanaged:
```

Comment:
Is not `""` meant to say "give me the default" - i.e. it is a valid state that means the same as `Managed`?

Reply:
there is not default and "" means unknown.

Reply:
@deads2k I thought you wanted to support rendering an empty object as valid?

Reply:
i would assume that for `ManagementState` we would always want to have an non-zero value and the field present in operator spec and this won't be "optional" field.

Reply:
https://github.com/openshift/api/blob/master/operator/v1/types.go#L46-L47

Reply:
@sttts ^

Reply:
@enj according to @sttts that has no effect, just docs and we should remove that. 

Using https://github.com/openshift/kubernetes-sigs-controller-tools/pull/4 we can specify pattern and only allow valid values to be passed.

Reply:
https://github.com/openshift/api/pull/232